docs/development/module/tests.md

```diff
@@ -6,104 +6,116 @@ Runtime tests allow you to verify the logic in your runtime module by mocking a
+To create the test environment in unit tests, the build method is called to generate a `TestExternalities` using the default genesis configuration. Then, [`with_externalities`](https://crates.parity.io/substrate_externalities/fn.with_externalities.html) provides the runtime environment in which we may call the module's methods to test that storage, events, and errors behave as expected.
 
 ```rust
-impl system::Trait for Test {
-	type Origin = Origin;
-	type Index = u64;
-	type BlockNumber = u64;
-	type Hash = H256;
-	type Call = ();
-	type Hashing = BlakeTwo256;
-	type AccountId = u64;
-	//--snip--
+#[test]
+fn fake_test_example() {
+	ExtBuilder::build().execute_with(|| {
+		// ...test conditions...
+	}) 
 }
 ```
 
-Here you can see that a core runtime type like `AccountId` can simply be set to `u64` rather than using real cryptographic primitives like `SR25519`. This means that to access accounts in your test, you simply need to pass an integer rather than more complicated stuff.
+Custom implementations of [Externalities](https://crates.parity.io/substrate_externalities/index.html) allow developers to construct runtime environments that provide access to features of the outer node. Another example of this can be found in [`substrate-offchain`](https://crates.parity.io/substrate_offchain/), which maintains its own [Externalities](https://crates.parity.io/substrate_offchain/testing/index.html) implementation. [Implementing configurable externalities](https://substrate.dev/recipes/testing/externalities.htm) is covered in more depth in the recipes.
 
-In cases where you do not need to use a type, like `Call` in the example above, you can pass the unit type `()`.
+#### Genesis Config
 
-### Mock Runtime Storage
+The previously shown `ExtBuilder::build()` method used the default genesis configuration for building the mock runtime environment. In many cases, it is convenient to set storage before testing.
 
-The `runtime-io` crate exposes two helpers for establishing storage for tests:
+An example might involve pre-seeding account balances before testing. The `TestRuntime` implements `balances::Trait`, 
 
-* `TestExternalities`: An in-memory, hashmap-based externalities implementation. In other words, it mocks a test storage needed for the runtime to execute in a minimal fashion.
+```rust
+parameter_types! {
+	pub const ExistentialDeposit: u64 = 1;
+	pub const TransferFee: u64 = 0;
+	pub const CreationFee: u64 = 0;
+}
 
-* `with_externalities`: A function that allows you to run logic on top of a `TestExternalities` storage.
+impl balances::Trait for TestRuntime {
+	type Balance = u64;
+	type OnNewAccount = ();
+	type OnFreeBalanceZero = ();
+	type Event = ();
+	type TransferPayment = ();
+	type DustRemoval = ();
+	type ExistentialDeposit = ExistentialDeposit;
+	type TransferFee = TransferFee;
+	type CreationFee = CreationFee;
+}
```

Comment:
I cant make a github suggestion here, but I think we should just keep this minimal:

```
impl balances::Trait for TestRuntime {
	type Balance = u64;
        //--snip//
}
```

All the other parts of the code seem irrelevant for the example.

Reply:
good call, recently addressed in last commit